src/mc_cell_group.hpp

```diff
@@ -72,32 +79,41 @@ class mc_cell_group: public cell_group {
 
     void reset() override {
         spikes_.clear();
-        events_.clear();
         reset_samplers();
-        binner_.reset();
+        for (auto& b: binners_) {
+            b.reset();
+        }
         lowered_.reset();
+        for (auto& lanes: event_lanes_) {
+            for (auto& lane: lanes) {
+                lane.clear();
+            }
+        }
     }
 
     void set_binning_policy(binning_kind policy, time_type bin_interval) override {
-        binner_ = event_binner(policy, bin_interval);
+        binners_.clear();
+        binners_.resize(gids_.size(), event_binner(policy, bin_interval));
```

Comment:
While the binner memory overhead is going to be similar (smaller but same order) in the `binning_kind::following` case, we're making many more objects in the `binning_kind::none` and `binning_kind::regular` cases. Is the change to make a separate binner for each cell a performance optimization for the `following` case?

Reply:
You are right. I missed this detail when refactoring.
The memory overheads aren't much, with one small object per cell. Given that this type is used only for multi-compartment cells, that is relatively little.

Reply:
If there's no significant performance gain from having one per cell, then we could just revert to the original binner, no?

Reply:
There is a performance penalty to using the old approach with the `following` case: a hash table lookup on every single event.